### DISCUSSION TITLE
Is it possible to sort the collection?

### DISCUSSION THREAD

**Discussion by BryanJBryce** [pointfreeco/swift-identified-collections#57] [Q&A]
I try to sort and it gets turned back into a normal array.

**Comment by mbrandonw** [comment]
You can use the in-place mutation version, [`sort(by:)`](https://pointfreeco.github.io/swift-identified-collections/main/documentation/identifiedcollections/identifiedarray/sort(by:)-36swl). By default, `Sequence` defines `sorted` as returning an array, so that kind of forces our hand here. We could technically overload `sorted` on `IdentifiedArray` to return another `IdentifiedArray`, but it wouldn't work generically. 

For example, even if we defined a new sorted to return `IdentifiedArray`, it will be the case that if you erase the type to `any Sequence` and then call `sorted` you will always get an `Array`:

```swift
var xs = IdentifiedArray<String, String>(id: \.self)
type(of: xs.sorted())  // IdentifiedArray<String, String>

var ys = xs as any Sequence<String>
type(of: ys.sorted())  // Array<String>
```

That behavior can be surprising, which I think is why we chose not to define `sorted`.

**Reply by mbrandonw** [reply]
Hey @BryanJBryce, after chatting with Stephen we think this could be a good thing to add to the library. Do you want to take a crack at it by opening a PR?
